public/components/facet/facet.js

```diff
@@ -38,36 +43,55 @@ export default ['$http', function($http) {
         if (search && search._id.length) scope.open = true;
       }, true)
 
+      /*
+       *  This function is called when the user clicks a checkbox.
+       *
+       *  In "or" mode,
+       *  - a CHECKED box is "selected" and "!negative"
+       *  - an UNCHECKED box is "!selected" and "!negative"
+       * 
+       *  In "and" mode,
+       *  - a NEGATIVE box is "selected" and "negative"
+       */
       scope.refresh = function(u) {
-        if (!u.selected && !u.negative && scope.operator !== "or") {
-          u.negative = true;
-          u.selected = true;
-        }
-        else if (!u.selected && u.negative) {
-          u.negative = false;
+        if (scope.loading) { // cannot refresh this unique until previous unique has finished
+          u.selected = false;
+          setTimeout(scope.refresh(u), 100);
```

Comment:
FYI, when doing setTimeout, the way you did it makes `scope.refresh(u)` evaluate immediately, thus giving this error:

![image](https://user-images.githubusercontent.com/1689183/62426744-a7a15680-b69d-11e9-81d0-90cedf070dc5.png)

The way to make sure that `scope.refresh(u)` evaluates after 100 ms is to wrap it in an enclosing function:

```suggestion
          setTimeout(() => scope.refresh(u), 100);
```

Actually, though, it seems like removing this line altogether keeps it working, so you can probably remove it.

Reply:
Interesting find!